gitpod-io/gitpod
# support integration of the supervisor frontend into the IDE top window

#### What it does

Gets rid of running IDE in a frame, there are different problems with it:
- not all required APIs are available for nested browsing contexts like keyboard api
- navigation is hard, since one should mutate query param, in some cases it is even impossible like going to website which does not allow embedding

Now IDE has to provide index.html file which includes supervisor frontend as a first script.

#### How to test

- start a regular workspace: http://ak-move-ide-to-top.staging.gitpod-dev.com/#https://github.com/gitpod-io/go-gin-app
- stop a workspace from the workspace and from the dashboard
- restart a workspace from the workspace and from the dashboard
- check navigation:
  - `File`-> `Open Workspace` should open a new window
  - enable preserve window preference and do `File`-> `Open Workspace`, it should reload the current window
  - click some link in the markdown the new window should be opened
  - click the gitpod logo the current window should be reloaded
- start a prebuild workspace: http://ak-move-ide-to-top.staging.gitpod-dev.com/#prebuild/https://github.com/gitpod-io/go-gin-app
